src/components/MultiHopTrade/components/TradeConfirm/TradeFooterButton.tsx

```diff
@@ -211,6 +216,33 @@ export const TradeFooterButton: FC<TradeFooterButtonProps> = ({
   const activeQuoteErrors = useAppSelector(selectActiveQuoteErrors)
   const activeQuoteError = useMemo(() => activeQuoteErrors?.[0], [activeQuoteErrors])
 
+  const isButtonLoading = useMemo(() => {
+    if (!confirmedTradeExecutionState) return true
+
+    return (
+      isSubmitting ||
+      confirmedTradeExecutionState === TradeExecutionState.Initializing ||
+      tradeButtonProps?.isLoading ||
+      isLoading ||
+      (swapperName === SwapperName.Zrx &&
+        isPermit2 &&
+        !tradeQuoteStep?.permit2Eip712 &&
+        ![TradeExecutionState.Initializing, TradeExecutionState.Previewing].includes(
+          confirmedTradeExecutionState,
+        ) &&
+        !activeQuoteError)
+    )
+  }, [
+    isSubmitting,
+    confirmedTradeExecutionState,
+    tradeButtonProps?.isLoading,
+    isLoading,
+    isPermit2,
+    activeQuoteError,
+    tradeQuoteStep?.permit2Eip712,
+    swapperName,
+  ])
+
```

Comment:
I hope this is the right home, I couldn't find a better place for this

Reply:
Sounds good to me, previous logic was already relying on the state machine loading logic + form state + passed loading state, this just adds ZRX-specific bits to it!